internal/server/statvar/statvar_group.go

```diff
@@ -116,20 +115,29 @@ func filterSVG(in *pb.StatVarGroups, statVars []string) *pb.StatVarGroups {
 	return result
 }
 
-// adjustDescendentStatVarCount returns the DescendentStatVarCount for node curID.
-// Mutates DescendentStatVarCount for all nodes in id2Node starting at customSvgRoot.
-func adjustDescendentStatVarCount(id2Node map[string]*pb.StatVarGroupNode, curID string) int32 {
+// adjustDescendentStatVarCount returns all the unique stat var under curID.
+// Recursively mutates DescendentStatVarCount for all nodes in id2Node.
+func adjustDescendentSVCount(
+	id2Node map[string]*pb.StatVarGroupNode, curID string,
+) map[string]struct{} {
 	curNode, ok := id2Node[curID]
 	if !ok {
-		return 0
+		return nil
+	}
+	descendentSV := map[string]struct{}{}
+	for _, sv := range curNode.ChildStatVars {
+		descendentSV[sv.Id] = struct{}{}
 	}
-	curNode.DescendentStatVarCount = int32(len(curNode.ChildStatVars))
 	for _, childSVG := range curNode.GetChildStatVarGroups() {
+		childDescendentSV := adjustDescendentSVCount(id2Node, childSVG.GetId())
```

Comment:
Each recursion creates multiple levels of the id->child map in memory and I'm afraid this is O(d * len(c)), d=depth and c being the max child per node. Could this cause OOM on a container?

Alternatively, descendentSV := map[string]struct{}{} can be declared before the first recursion call and descendentSV can be passed down to every recursive call.

Reply:
I think the set from a parent SVG can not be passed to all the child SVG, so each child would always need to create its own set (and hopefully GC will kick in when the recursion returns from children SVG).